### DISCUSSION TITLE
Model processors vs. directly calling function on parsed model

### DISCUSSION THREAD

**Discussion by dkrikun** [textX/textX#334] [Q&A]
Question: how is using a model processor is different from simply calling a function taking a model *after* the parsing is complete?

**Comment by igordejanovic** [comment]
No difference. Model processor is only a convenience that ensures that calling a processor is not forgotten.

**Comment by igordejanovic** [comment]
I think that we should not close `question` issues to make them more visible.

**Comment by dkrikun** [comment]
By the way, are object processors also called after the parsing is complete? If an object processor returns a value of another object type, does a processor for that type is called?

**Comment by goto40** [comment]
Object processors are called after parsing and reference resolution. Thus, an object processor always have all references resolved. 

If you return a value in an object processor, the current object is replaced. The replaced object is not further processed.

Details: see call_obj_processors in model.py.

**Comment by goto40** [comment]
> I think that we should not close `question` issues to make them more visible.

Can we introduce an "answered question" . This will make it easier to "see" what issues are "open" or not...
